### ISSUE TITLE
some data retrieval, migration and validation code could be moved out of nmdc-schema

### ISSUE THREAD

**Issue by turbomam** [microbiomedata/nmdc-schema#1665]
identified in nmdc-schema's `project.Makefile` and `pyproject.toml`:

- `build-datafile-from-api-requests` = "nmdc_schema.build_datafile_from_api_requests:main"
  - could probably be deprecated in favor of `get-study-related-records`
- `get-study-related-records` = "src.scripts.nmdc_database_tools:cli"
  - project.Makefile is only using the `extract-study` subcommand
  - might not be a good CLI name, because nmdc_database_tools was written to take more subcommands
- `migration-recursion` = 'nmdc_schema.migration_recursion:main'
- `pure-export` = "nmdc_schema.mongo_dump_api_emph:cli"
- `class-sparql` = "nmdc_schema.class_sparql:main"
  - retrieves data from a triplestore. may have some old harcoded configuration. may not be the best way to way to illustrate sparql as an nmdc solution, but does have nice use cases. now experimenting with local TDB2 and Fuseki

@turbomam make notes of which resources they query from, like PyMongo or runtime API. @mbthornton-lbl has spoken of a _nmdc-common_ or _nmdc-client_ repo which would manage all access via unauthenticated API calls. Not all of the tooling mentioned above would meet that criterion.

We could also link this to documentation bout the query, find and metadata API endpoints.

**Comment by turbomam** [comment]
class-sparql returns CURIes not URIs

**Comment by eecavanna** [comment]
I'm one of the assignees of this issue. I don't understand my role with respect to this issue.

**Comment by turbomam** [comment]
> I'm one of the assignees of this issue. I don't understand my role with respect to this issue.

OK, I'll clarify by Monday or remove you,

**Comment by turbomam** [comment]
The scripts listed here — `pure-export`, `migration-recursion`, `build-datafile-from-api-requests` — are the same ones targeted for removal in #2969 once `linkml validate` can read MongoDB directly (linkml/linkml#3380).

**Comment by turbomam** [comment]
superseded by or gated on
- https://github.com/microbiomedata/nmdc-schema/issues/2969
- https://github.com/linkml/linkml/issues/3380
